src/ucp/tag/rndv.c

```diff
@@ -274,13 +329,18 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_rma_get_zcopy, (self),
                    length, rndv_req->send.lane);
 
     state = rndv_req->send.state.dt;
+    /* TODO: is this correct? memh array may skip MD's where
+     * registration is not supported. for now SHM may avoid registration,
```

Comment:
so maybe ucp_dt_iov_copy_uct() should get md_index instead of memh_idx, check if it requires registration, and only if yes, call ucp_memh_map2idx()?

Reply:
it will make copy_uct function too complex (at least we have to add there context, and may be request itself), it have to know what king of operation is processing (send/recv) to select appropriate registration method, etc... it is already over complicated, but it is under risk became nightmare

Reply:
I updated parameter to pass md index (instead of memh index)

Reply:
so if there is no uct_memh (md does not have registration), does it mean it can access un-initialized memory?

Reply:
as I can see it is possible for SHM only, in this case multi-lane is not used and memh is not used in RMA operations

Reply:
yep, it is possible to get uninitialized memory (memh value), I don't see code where it is set to NULL